### ISSUE TITLE
Metric used by ModelTrainer.train() 'patience' argument

### ISSUE THREAD

**Issue by Flowhill** [flairNLP/flair#2926]
In this issue [#864](https://github.com/flairNLP/flair/issues/864) it becomes clear that flair uses an Early Stopping mechanism by default.
That Early Stopping mechanism seems to be determined by the parameter 'main_evaluation_metric':
https://github.com/flairNLP/flair/blob/493b61fc07f83d412928301571b0e3abe780348f/flair/trainers/trainer.py#L163
The mechanism lowers the learning rate when the chosen metric stops improving, eventually stopping when: learning rate < min_learning_rate

Another parameter that could induce early stopping is 'patience'
https://github.com/flairNLP/flair/blob/493b61fc07f83d412928301571b0e3abe780348f/flair/trainers/trainer.py#L139

However it is not clear to me whether patience uses the 'main_evaluation_metric' or simply the loss on the training set.

**Comment by alanakbik** [comment]
Hello @Flowhill, the `patience` decides how many "bad epochs" (of no improvement) it waits before annealing the learning rate. By default it anneals against the main_evaluation_metric determined on the dev data. Only if there is no dev data (for instance if you set `train_with_dev=True`) it will anneal against training loss.

**Comment by Flowhill** [comment]
Thnks @alanakbik 100% clear.
